src/coreclr/vm/eetwain.cpp

```diff
@@ -4445,7 +4445,7 @@ FCIMPL1(void, GCReporting::Unregister, GCFrame* frame)
 
     // Destroy the GCFrame.
     _ASSERTE(frame != NULL);
-    frame->Pop();
+    frame->Remove();
```

Comment:
The `Push()`/`Pop()` names were introduced intentionally to provide naming symmetry. Is there a reason the `Pop()` can't be reimplemented as `Remove()`? If the `Remove()` is really so fundamentally different and we need the existing `Pop()`, I would instead call it `PopAll()` or something that aligns with the `Push()` above in `GCReporting::Register()`.

Reply:
Please note that this is not replacing `Pop` for the GC frame chain in general. This `GCReporting::Unregister` is a special case when with the new exception handling enabled, there can be GC frames pushed after the one pushed by the `GCReporting::Register`. I'd prefer keeping the `Push` / `Pop` for all the other cases, as it verifies that we are not popping out of order. `PopAll` sounds like we would be popping all frames until the passed in frame. I see the operations of `Push` /  `Pop` / `Remove` as operations on the underlying linked list

Reply:
> `Remove` as operations on the underlying linked list

Ah, so this really is a remove this particular frame from anywhere in the call stack/frame chain. I retract my comment. This makes sense.